### PR TITLE
717 Default to use current activations course

### PR DESCRIPTION
**Pull request by abidingotter**

### CODE REVIEW

File: Source/Plugins/Core/com.equella.core/resources/lang/i18n-resource-centre.properties [hunk 1]
```diff
@@ -4479,6 +4479,7 @@ schemas.title=Schemas
 api.error.dateparse=Failed to parse date {0}
 
 component.courseselectionlist.selectedcourse={0} - {1}
+component.courseselectionlist.selectedcoursenocode={0}
```
Comment (doolse) [inline Source/Plugins/Core/com.equella.core/resources/lang/i18n-resource-centre.properties]: Why even have a lang string for this?

File: .idea/.gitignore [hunk 2]
```diff
@@ -33,4 +33,7 @@
 /hydra.xml
 /inspectionProfiles/Project_Default.xml
 
+/google-java-format.xml
```
Comment (doolse) [inline .idea/.gitignore]: We need this in the repo

Comment (abidingotter) [reply .idea/.gitignore]: I will let you deal with that

File: Source/Plugins/Core/com.equella.core/src/com/tle/web/activation/operation/BulkRolloverOperation.java [hunk 3]
```diff
@@ -192,10 +195,18 @@ public String getTitleKey() {
         rolloverDates);
   }
 
+  private CourseInfo getSelectedCourse(SectionInfo info) {
+    final CourseInfo course = courses.getSelectedValue(info);
+    if (course != null && course.getCode().equals("existing")) {
```
Comment (doolse) [inline Source/Plugins/Core/com.equella.core/src/com/tle/web/activation/operation/BulkRolloverOperation.java]: What if the course's code was "existing"?

File: Source/Plugins/Core/com.equella.core/resources/web/scripts/component/select2helper.js [hunk 4]
```diff
@@ -1,72 +1,93 @@
 var select2helper = {};
 
+/**
```
Comment (doolse) [inline Source/Plugins/Core/com.equella.core/resources/web/scripts/component/select2helper.js]: Pls remove commented code

Comment (abidingotter) [reply Source/Plugins/Core/com.equella.core/resources/web/scripts/component/select2helper.js]: This is doc, not commented code.  Added further comments to make it obvious

File: Source/Plugins/Core/com.equella.core/resources/lang/i18n-resource-centre.properties (hunk 1 above)
Comment (abidingotter) [reply Source/Plugins/Core/com.equella.core/resources/lang/i18n-resource-centre.properties]: Gone

File: Source/Plugins/Core/com.equella.core/src/com/tle/web/activation/operation/BulkRolloverOperation.java (hunk 3 above)
Comment (abidingotter) [reply Source/Plugins/Core/com.equella.core/src/com/tle/web/activation/operation/BulkRolloverOperation.java]: It should be checking the UUID.  Good catch.

File: .idea/.gitignore (hunk 2 above)
Comment (abidingotter) [reply .idea/.gitignore]: Have removed from ignore and the commit

File: Source/Plugins/Core/com.equella.core/resources/web/scripts/component/select2helper.js [hunk 5]
```diff
@@ -1,72 +1,99 @@
...
+          return params.ajaxurl;
+        },
+        data: function(ajaxParams) {
+          var start = ajaxParams.page
+            ? (ajaxParams.page - 1) * perpage
+            : undefined;
+          var queryParams = {
+            q: ajaxParams.term,
+            length: perpage,
+            resumption: start ? start + ":" + (start + perpage) : undefined
```
Comment (doolse) [inline Source/Plugins/Core/com.equella.core/resources/web/scripts/component/select2helper.js]: You can't generate your own resumption tokens, you have to use the one you get back from the previous request

**Pull request merged**